add Interpretation.show_at method

vrodriguezf opened the pull request:
This is basically the code added (by @muellerzr)

```python
def __getitem__(self:Interpretation, idxs):
    "Get the inputs, preds, targets, decoded outputs, and losses at `idx`"
    if not is_listy(idxs): idxs = [idxs]
    attrs = 'inputs,preds,targs,decoded,losses'
    res = L([getattr(self, attr)[idxs] for attr in attrs.split(',')])
    return res

@patch
def show_at(self:Interpretation, idx:int, **kwargs):
    inp, _, targ, dec, _ = self[idx]
    dec, targ = L(dec, targ).map(Self.unsqueeze(0))
    self.dl.show_results((inp, dec), targ, **kwargs)
```

I just showcased it following in a nb following the format of this repo. The methods are exported to the module `basics.interp.py`

muellerzr commented:
Thanks @vrodriguezf (finally able to sit down and take a look). It would be _nice_ if we could explain just what our `show_at` code is doing internally, as we both know there's much to be learned through those simple lines of code! Would it be reasonable to ask you to make that a bit more verbose? 

If that's a bit out of your ballpark, I can expand upon it later, just wanted to give you the opportunity first!

Also, when building make sure you have the latest pip nbdev (1.1.11) and do `nbdev_build_lib`. This _should_ fix your build errors.

Thank you again! :)

muellerzr commented on wwf/basics/interp.py:
```diff
@@ -0,0 +1,18 @@
+    return res
+
+@patch
+def show_at(self:Interpretation, idx:int, **kwargs):
```
Perhaps include a docstring here. Also would we want to have them be able to show multiple index's? Or is just one good enough. I'm leaning towards a situation similar to `__getitem__` where we check for a list of index's

vrodriguezf replied:
Sure! I assume that the creation of the grid for showing multiple items would be delegated to `show_results`, is that correct?

muellerzr replied:
Yes that would be (sorry for just now getting to that)

muellerzr reviewed (changes requested):
Also let's try to rename the notebook `05_interp.show_at`, this way anything else tying to interp goes to `interp.x`. I think I'm fine with the basics, I like having that there and may move over the `lr_finder` improvements to that as well! It's a good way to section out what's an improvement vs an implementation.

vrodriguezf commented:
Will do @muellerzr , thanks for the review!

muellerzr commented:
Correct. For it to work I believe you just need to modify idxs to become a
list (similar to the first two lines in the other function). Otherwise the
rest should still be the exact same

On Fri, Jan 15, 2021 at 3:22 AM Victor Rodriguez-Fernandez <
notifications@github.com> wrote:

> *@vrodriguezf* commented on this pull request.
> ------------------------------
>
> In wwf/basics/interp.py
> <https://github.com/walkwithfastai/walkwithfastai.github.io/pull/20#discussion_r558111111>
> :
>
> > @@ -0,0 +1,18 @@
> +# AUTOGENERATED! DO NOT EDIT! File to edit: nbs/05_basics.interp.show_at.ipynb (unless otherwise specified).
> +
> +__all__ = []
> +
> +# Cell
> +@patch
> +def __getitem__(self:Interpretation, idxs):
> +    "Get the inputs, preds, targets, decoded outputs, and losses at `idx`"
> +    if not is_listy(idxs): idxs = [idxs]
> +    attrs = 'inputs,preds,targs,decoded,losses'
> +    res = L([getattr(self, attr)[idxs] for attr in attrs.split(',')])
> +    return res
> +
> +@patch
> +def show_at(self:Interpretation, idx:int, **kwargs):
>
> Sure! I assume that the creation of the grid for showing multiple items
> would be delegated to show_results, is that correct?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/walkwithfastai/walkwithfastai.github.io/pull/20#discussion_r558111111>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB3YCV6JYEBHFUVAHUDZAJ3S2ACNVANCNFSM4V4LU3NQ>
> .
>

vrodriguezf commented:
Hi! Sorry for the delay, it was a crazy week...I updated to nbdev 1.1.11 and added some explanations to the code of `show_at`. It's still failing though!

muellerzr commented:
Awesome work @vrodriguezf! It's not failing for me, so don't worry about that :) Thank you for the hard work and an awesome post!
